File: examples/chat-room-python/scripts/connect.py
import asyncio
import os
from actor_core_client import AsyncClient as ActorClient

async def main():
    # Create client
    endpoint = os.getenv("ENDPOINT", "http://localhost:6420")
    client = ActorClient(endpoint)

    # Connect to chat room
    chat_room = await client.get("chatRoom")

    # Get existing messages
    messages = await chat_room.action("getHistory", [])
    print("Messages:", messages)

    # Listen for new messages
    def on_message(username: str, message: str):
        print(f"Message from {username}: {message}")

    chat_room.on_event("newMessage", on_message)

    # Send message to room
    await chat_room.action("sendMessage", ["william", "All the world's a stage."])

    # Disconnect from actor when finished
    await chat_room.disconnect()

if __name__ == "__main__":
    asyncio.run(main()) 

File: examples/chat-room/src/scripts/connect.ts
/// <reference types="node" />
import { createClient } from "@rivetkit/actor/client";
import type { registry } from "../backend/registry";

async function main() {
	// Create type-aware client
	const client = createClient<typeof registry>(
		process.env.ENDPOINT ?? "http://localhost:8080",
	);

	// connect to chat room
	const chatRoom = client.chatRoom.getOrCreate().connect();

	// call action to get existing messages
	const messages = await chatRoom.getHistory();
	console.log("Messages:", messages);

	// listen for new messages
	chatRoom.on("newMessage", (message: any) =>
		console.log(`Message from ${message.sender}: ${message.text}`),
	);

	// send message to room
	await chatRoom.sendMessage("william", "All the world's a stage.");

	// disconnect from actor when finished
	await chatRoom.dispose();
}

main();
